scripts/pre-commit

```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python
+#
+#  Copyright (C) 2022
+#  MIT
+#
+#
+#  This program is free software; you can redistribute it and/or modify
+#  it under the terms of the GNU General Public License as published by
+#  the Free Software Foundation; either version 3 of the License, or
+#  (at your option) any later version.
+#
+#  This program is distributed in the hope that it will be useful,
+#  but WITHOUT ANY WARRANTY; without even the implied warranty of
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License along
+#  with this program; if not, write to the Free Software Foundation, Inc.,
+#  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+'''Git pre-commit hook to update the copyright year in Sherpa code files
+'''
+from __future__ import print_function
```

Comment:
We shouldn't need this, should we?

Reply:
git pre-commit scripts are run with whatever the active Python is in your development environment. Usually, I have a Python 3 active through conda, but my system Python is still 2.7, so we want this script to work in those cases, too. (I don't think it's too crucial, since only developers who know what's happening will use this feature, but it does not hurt to have it.)